src/pages/day-view.tsx

```diff
@@ -0,0 +1,42 @@
+import type { NextPage } from "next";
+import { TodoItem } from "../features/todo/list";
+import { Title } from "../features/ui/title";
+import { trpc } from "../utils/trpc";
+
+const DayViewPage: NextPage = () => {
+    const inboxQuery = trpc.planning.inbox.useQuery();
+    const todos = inboxQuery.data?.todos;
+    const todoGroupedByGoal = todos?.reduce((acc, todo) => {
+        const goal = todo.goalName;
+        if (!acc[goal] || !Array.isArray(acc[goal])) {
+            acc[goal] = [];
+        }
+        else {
+            acc[goal]?.push(todo);
+        }
+        return acc;
+    }, {} as Record<string, typeof todos>);
+
+    if (todoGroupedByGoal === undefined) {
+        return <div>Loading...</div>;
```

Comment:
It's better to user `Loader` component, looks a bit nicer (also easier to change in future)